File: BE/Boek.Infrastructure/ViewModels/BookProducts/Mobile/SubHierarchicalBookProductsViewModel.cs
using Boek.Infrastructure.ViewModels.Genres;
using Boek.Infrastructure.ViewModels.Users;

namespace Boek.Infrastructure.ViewModels.BookProducts.Mobile
{
    public class SubHierarchicalBookProductsViewModel
    {
        public string SubTitle { get; set; }
        public int? GenreId { get; set; }
        public Guid? IssuerId { get; set; }
        public GenreViewModel Genre { get; set; }
        public UserViewModel Issuer { get; set; }
        public List<MobileBookProductsViewModel> BookProducts { get; set; }
    }
}

File: Mobile/objects/viewmodels/BookProduct/SubHierarchicalBookProductsViewModel.ts
import { GenreViewModel } from "../Genres/GenreViewModel";
import { UserViewModel } from "../Users/UserViewModel";
import { MobileBookProductViewModel } from "./Mobile/MobileBookProductViewModel";

export interface SubHierarchicalBookProductsViewModel {
    subTitle: string;
    genreId?: number;
    issuerId?: string;
    genre?: GenreViewModel;
    issuer?: UserViewModel;
    bookProducts?: MobileBookProductViewModel[];
}